packages/vuetify/src/components/VProgressLinear/VProgressLinear.ts

```diff
@@ -105,82 +157,72 @@ export default mixins(Colorable).extend({
       }
 
       return styles
     }
   },
 
   methods: {
-    genDeterminate (h: CreateElement): VNode {
-      return h('div', this.setBackgroundColor(this.color, {
-        ref: 'front',
-        staticClass: `v-progress-linear__bar__determinate`,
-        style: {
-          width: `${this.effectiveWidth}%`
-        }
-      }))
+    genBuffer () {
+      return this.$createElement('div', {
+        staticClass: 'v-progress-linear__buffer',
+        style: this.styles
+      })
     },
-    genBar (h: CreateElement, name: string): VNode {
-      return h('div', this.setBackgroundColor(this.color, {
-        staticClass: 'v-progress-linear__bar__indeterminate',
+    genContent () {
+      const slot = getSlot(this, 'default', { value: this.internalLazyValue })
+
+      if (!slot) return null
+
+      return this.$createElement('div', {
+        staticClass: 'v-progress-linear__content'
+      }, slot)
+    },
+    genProgressBar (name: 'long' | 'short') {
+      return this.$createElement('div', this.setBackgroundColor(this.color, {
+        staticClass: 'v-progress-linear__indeterminate',
         class: {
           [name]: true
         }
       }))
-    },
-    genIndeterminate (h: CreateElement): VNode {
-      return h('div', {
-        ref: 'front',
-        staticClass: 'v-progress-linear__bar__indeterminate',
-        class: {
-          'v-progress-linear__bar__indeterminate--active': this.active
-        }
-      }, [
-        this.genBar(h, 'long'),
-        this.genBar(h, 'short')
-      ])
     }
   },
 
   render (h): VNode {
-    const fade = h(VFadeTransition, this.indeterminate ? [this.genIndeterminate(h)] : [])
-    const slide = h(VSlideXTransition, this.indeterminate ? [] : [this.genDeterminate(h)])
-
-    const bar = h('div', {
-      staticClass: 'v-progress-linear__bar',
-      style: this.styles
-    }, [fade, slide])
-
-    const background = h('div', this.setBackgroundColor(this.backgroundColor || this.color, {
-      staticClass: 'v-progress-linear__background',
-      style: this.backgroundStyle
-    }))
-
-    const content = this.$slots.default && h('div', {
-      staticClass: 'v-progress-linear__content'
-    }, this.$slots.default)
-
-    return h('div', {
+    const data = {
       staticClass: 'v-progress-linear',
       attrs: {
         'role': 'progressbar',
         'aria-valuemin': 0,
         'aria-valuemax': this.normalizedBufer,
         'aria-valuenow': this.indeterminate ? undefined : this.normalizedValue
       },
-      class: {
-        'v-progress-linear--query': this.query
-      },
+      class: this.classes,
       style: {
-        height: convertToUnit(this.height)
-      },
-      on: this.$listeners
-    }, [
-      background,
-      bar,
-      content
+        bottom: this.bottom ? 0 : undefined,
+        height: this.active ? convertToUnit(this.height) : 0,
+        top: this.top ? 0 : undefined
+      }
+    }
+
+    if (this.reactive) {
+      this._g(data, {
```

Comment:
Why use this internal API method instead of something like

```
data.on = {
  click: () => {}
}
```

or if you explicitly need to merge listeners (which is not the case here)

```
data.on = {
  ...data.on,
  click: () => {}
}
```